search-impl/src/test/java/com/example/core/TopicStub.java

```diff
@@ -49,13 +54,21 @@ public TopicStub(Materializer materializer) {
                         .via(flow)
                         .toMat(Sink.ignore(), Keep.both())
                         .run(materializer);
-                sourceActor = pair.first();
+                sourceActor.complete(pair.first());
                 return pair.second();
```

Comment:
```java
Source.actorRef(1, OverflowStrategy.fail())
```

I think the above is too low a buffer - you'd only have to send twice in a row and you'd risk the stream failing.  It's safe to set it at something quite high, maybe 1024.